qiskit/visualization/latex.py

```diff
@@ -594,42 +594,75 @@ def _add_condition(self, op, wire_list, col):
         #         or if cregbundle, wire number of the condition register itself
         # gap - the number of wires from cwire to the bottom gate qubit
 
-        creg_size = self.cregs[op.condition[0]]
-        if_value = format(op.condition[1], "b").zfill(creg_size)
-        if not self.reverse_bits:
-            if_value = if_value[::-1]
+        cond_is_bit = isinstance(op.condition[0], Clbit)
+        if cond_is_bit:
+            cond_reg = self.bit_locations[op.condition[0]]["register"]
+        else:
+            cond_reg = op.condition[0]
+        creg_size = self.cregs[cond_reg]
+        if cond_is_bit:
+            if_value = op.condition[1]
+        else:
+            if_value = format(op.condition[1], "b").zfill(creg_size)
+        if not cond_is_bit:
+            if not self.reverse_bits:
+                if_value = if_value[::-1]
```

Comment:
```
        cond_is_bit = isinstance(op.condition[0], Clbit)
        creg_size = self.cregs[cond_reg]
        if cond_is_bit:
            cond_reg = self.bit_locations[op.condition[0]]["register"]
            if_value = op.condition[1]
        else:
            cond_reg = op.condition[0]
            if_value = format(op.condition[1], "b").zfill(creg_size)
            if not self.reverse_bits:
                if_value = if_value[::-1]
```
This might be a bit more compact.